server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1374,9 +1374,8 @@ public void loadIntoContext(ShardSearchRequest request, SearchContext context, Q
             queryPhase.execute(context);
             try {
                 context.queryResult().writeToNoId(out);
-
             } catch (IOException e) {
-                throw new AssertionError("Could not serialize response", e);
+                throw new IllegalArgumentException("Could not serialize response", e);
```

Comment:
This should not be needed anymore, ? We should keep the error as it is.

Reply:
I think this statement is unsafe.

Reply:
Then it should be an `UncheckedIOException` but imo this change is not needed.

Reply:
In general when there is an IOException, you're in big trouble anyway, and trying to handle it is quite heroic. Since the signature of this method throws Exception, I'd be in favor of removing this catch block too. The root cause of the problem is that StreamOutput throws an IOException for something that is not I/O related.